### DISCUSSION TITLE
How can I apply Tailwind CSS classes to SVG elements using D3.js?

### DISCUSSION THREAD

**Discussion by Laaouatni** [tailwindlabs/tailwindcss#10747] [Help] [answered]
I have just answer a StackOverflow question on this topic here: https://stackoverflow.com/questions/75666529/how-can-i-apply-tailwind-css-classes-to-svg-elements-using-d3-js/75677870#75677870

and seems to work fine as intended (see link), 
<br>but the only problem is the vscode extension that doesn't shows suggestion when using `.classed()` in D3js.

the `.classed()` is inside a javascript file. 
<br>the configuration of the tailwind is correct. 

I just wondering if is there a way to configure/add a regex to check also for the `.classed()` method in a javascript file. 
<br>if yes do I need to do it myself by changing the configuration or do I need to wait for the open-source contributors of the team TailwindCSS to release a new version.

------

example of photos of no popup of suggestion appears in vscode:

![image](https://user-images.githubusercontent.com/87947051/223844150-59bf3221-0c3c-426a-b98d-31ae84ec4ef8.png)
![image](https://user-images.githubusercontent.com/87947051/223844198-76da086a-c7d4-4f01-8a4c-494247fd8c02.png)
![image](https://user-images.githubusercontent.com/87947051/223844104-7fa14955-f729-4e19-a57a-197bb8e11f2c.png)
![image](https://user-images.githubusercontent.com/87947051/223844501-0f1570eb-4cb4-4de9-9f1b-84d61beb6d13.png)

I saw it isn't a problem of tailwind that can't read a js file since in this case the popup is shown:

![image](https://user-images.githubusercontent.com/87947051/223845042-335fc856-34a3-40ec-8b52-e8c7da6e86f4.png)

**Comment by wongjn** [comment, marked as answer]
You could look at [configuring the `tailwindCSS.experimental.classRegex` option](https://github.com/tailwindlabs/tailwindcss/issues/7553#issuecomment-735915659).

**Reply by Laaouatni** [reply]
thanks @wongjn, I edited my answer on stackoverflow with correct information. very cool configuration! :) 

-------

and here the regex I used (if someone in the future have the same problem):

```javascript
"tailwindCSS.experimental.classRegex": [
   "classed\\(\"([^)]*)\"\\)"
]
```
